core/node/node_framework/src/implementations/layers/web3_api/tx_sink/whitelist.rs

```diff
@@ -0,0 +1,98 @@
+use async_trait::async_trait;
+use zksync_config::configs::api::DeploymentAllowlist;
+use zksync_node_api_server::tx_sender::{
+    master_pool_sink::MasterPoolSink,
+    whitelist::{AllowListTask, WhitelistedDeployPoolSink},
+};
+
+use crate::{
+    implementations::resources::{
+        pools::{MasterPool, PoolResource},
+        web3_api::TxSinkResource,
+    },
+    service::StopReceiver,
+    task::{Task, TaskId, TaskKind},
+    wiring_layer::{WiringError, WiringLayer},
+    FromContext, IntoContext,
+};
+
+/// Wiring layer for [`WhitelistedDeployPoolSink`] that wraps a `MasterPoolSink` and enables allowlist filtering.
+pub struct WhitelistedMasterPoolSinkLayer {
+    pub deployment_allowlist: DeploymentAllowlist,
+}
+
+#[derive(Debug, FromContext)]
+#[context(crate = crate)]
+pub struct Input {
+    pub master_pool: PoolResource<MasterPool>,
+}
+
+#[derive(Debug, IntoContext)]
+#[context(crate = crate)]
+pub struct Output {
+    pub tx_sink: TxSinkResource,
+    #[context(task)]
+    pub allow_list_task: AllowListTask,
+}
+
+#[async_trait]
+impl WiringLayer for WhitelistedMasterPoolSinkLayer {
+    type Input = Input;
+    type Output = Output;
+
+    fn layer_name(&self) -> &'static str {
+        "whitelisted_master_pool_sink_layer"
+    }
+
+    async fn wire(self, input: Self::Input) -> Result<Self::Output, WiringError> {
+        let pool = input.master_pool.get().await?;
+        let master_pool_sink = MasterPoolSink::new(pool);
+
+        let allow_list_task = AllowListTask::from_config(self.deployment_allowlist);
+
+        let tx_sink =
+            WhitelistedDeployPoolSink::new(master_pool_sink, allow_list_task.clone()).into();
+
+        Ok(Output {
+            tx_sink,
+            allow_list_task,
+        })
+    }
+}
+
+#[async_trait]
+impl Task for AllowListTask {
+    fn id(&self) -> TaskId {
+        "api_allowlist_task".into()
+    }
+
+    fn kind(&self) -> TaskKind {
+        TaskKind::UnconstrainedTask
+    }
+
+    async fn run(self: Box<Self>, mut stop_receiver: StopReceiver) -> anyhow::Result<()> {
```

Comment:
Please move `run()` to the `api_server` method as well as an inherent method taking `self` receiver. Here, it should just delegate to it [as is the case for other components](https://github.com/matter-labs/zksync-era/blob/d1bac66e339e46d2ecac183db8d33d9ee3e85940/core/node/node_framework/src/implementations/layers/consistency_checker.rs#L103-L105). This will allow for better encapsulation, like making most of `AllowListTask` methods private.